### PR TITLE
development advices and common libraries

### PR DESCRIPTION
**Pull request by tolgacanunal**

### CODE REVIEW

File: BestPractices.md [hunk 1]
```diff
@@ -80,8 +80,53 @@ This chapter outlines best practices to use for developing in iOS and Android.
...
 
-### Common libraries
+### Architectures
+
+### Development Advices
 
-#### - Networking - Parsing
-#### - Image Download
-#### - Dependecy Injection
+* Try not to use nested fragments in activities.
```
Comment (ahmetmentes) [inline BestPractices.md]: I don't agree with this. What is the source of that statement?

Comment (tolgacanunal) [reply BestPractices.md]: Using fragment inside of fragments can make it hard to access UI elements in the activity. So, if title or anything else changes in the root activity, at least 2 listener needs to be added to each activity and fragments.

Comment (tolgacanunal) [reply BestPractices.md]: and also, I don't know if it's fixed or not but may face with the bug in the link http://delyan.me/android-s-matryoshka-problem/

File: BestPractices.md [hunk 2]
```diff
@@ -80,8 +80,53 @@ This chapter outlines best practices to use for developing in iOS and Android.
...
+* Do not put too much code in activities.
+* Code blocks that are used more than one activities should put into *Utility* classes.
```
Comment (yasin-ce) [inline BestPractices.md]: Or  into Base Classes.

File: BestPractices.md [hunk 3]
```diff
@@ -80,8 +80,53 @@ This chapter outlines best practices to use for developing in iOS and Android.
...
+* Rules of code guidelines must be followed while writing the code. 
+  * [**[*Java*]** Code Guideline](https://github.com/ahmetmentes/android-guidelines)
+  * [**[*Kotlin*]** Code Guideline](https://android.github.io/kotlin-guides/style.html)
+* Simple values that needs to be persisted in application can be put into SharedPreferences.
+* It's a good idea to make custom view attributes changeable from XML file.
+* *Do not use* hardcoded strings in layout and put all strings to **strings.xml** file.
+* All colors must be declared in **colors.xml** file to use.
+* Strings that are declared in **strings.xml** should be not in all caps. If you need to use in all caps, use textAllCaps feature.
```
Comment (yasin-ce) [inline BestPractices.md]: Why? If only all caps version is used in the whole app then you can write it as all caps. imho.

File: BestPractices.md (hunk 2 above)
Comment (tolgacanunal) [reply BestPractices.md]: I'll add it.

File: BestPractices.md (hunk 3 above)
Comment (yasin-ce) [inline BestPractices.md]: Let's also mention usage of "tools:" keyword.

Comment (tolgacanunal) [reply BestPractices.md]: You might need that text in the future as not all in caps. Of course, It's a preference not a must. and also, first letter capitalized strings are easier while translating as well. 
Should I delete this statement?

Comment (yasin-ce) [reply BestPractices.md]: Okay it makes sense but we can improve this like "If the exact same text has more than one version with different caps try to use textAllCaps to reduce the number of defined strings "

**Review by yasin-ce — APPROVED** [review]

**Pull request merged**